src/lsm/superblock_free_set.zig

```diff
@@ -175,12 +188,32 @@ pub const SuperBlockFreeSet = struct {
     }
 
     /// Returns the number of bytes written to `target`.
+    /// The encoded data does *not* incldue staged changes.
     pub fn encode(set: SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
         assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
 
         return ewah.encode(bitset_masks(set.blocks), target);
     }
 
+    /// Returns the number of bytes written to `target`.
+    /// The encoded data *does* include staged changes.
+    pub fn encode_with_staging(set: *SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
+        const count_free = set.count_released();
+        const count_staged = set.staging.count();
+        assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
+
+        // Temporarily mark the staged blocks as free.
+        set.blocks.toggleSet(set.staging);
+        assert(set.blocks.count() == count_free + count_staged);
+
+        const encode_size = ewah.encode(bitset_masks(set.blocks), target);
+        // Restore the changes: mark the staged blocks as allocated again.
+        set.blocks.toggleSet(set.staging);
+        assert(set.blocks.count() == count_free);
```

Comment:
If you like, what do you think about putting this in a defer block right where we do the toggle? Then the return line for this function can match that of `encode()`.